extras/jsonrpc/daemon.go

```diff
@@ -595,3 +601,42 @@ func (d *Client) AccountAdd(accountName string, seed *string, privateKey *string
 	structs.DefaultTagName = "json"
 	return response, d.call(response, "account_add", structs.Map(args))
 }
+
+type WalletCreateOpts struct {
+	SkipOnStartup bool
+	CreateAccount bool
+	SingleKey     bool
+}
+
+func (d *Client) WalletCreate(id string, opts *WalletCreateOpts) (*Wallet, error) {
```

Comment:
we followed up in slack. I think we need more opinions here.
Struct pros:
- easier to manage when not all params are necessary so that we don't have to specify nil each time
```
d.WalletCreate(id, nil)
vs
d.WalletCreate(id, nil, nil, nil, nil)
```

Cons:
* When the structure changes, those using the library will not necessarily notice and the behavior of the function might not reflect the intended purpose making it unclear as to what needs to be changed (tests should prevent that)

Reply:
It also makes it easier to omit optional parameters and supply only the ones we want:

```
d.WalletCreate(id, &WalletCreateOpts{CreateAccount: true})
// vs
d.WalletCreate(id, nil, nil, true, nil)
```

I think this also makes sense where we're designing an RPC client, which, in turn, relies on an API we don't control as a team. We don't want to break, say, `AccountList` for all clients just because there is a new `WalletID` parameter now that we need to use.

Reply:
Reposting from slack: 

to me it makes sense to use arguments for required params and an options struct for optional params

alternatively, if there's an option that's used a lot when calling the method, that can be an arg too

Reply:
@sayplastic I'm confused what you mean by 

> and break, say, AccountList for all clients just because there is a new WalletID parameter now that we need to use

How would we break AccountList for all clients? it would not break unless they update to a newer version of lbry.go. And if they are updating, they should be aware that there may be API changes that they'll have to handle. Nothing will break without them taking action.

Reply:
Not the best wording on my part. What I meant is we’re going to be requiring changes from client code if they stay up to date.